llvm/test/Transforms/LoopVectorize/conditional-scalar-assignment-vplan.ll

```diff
@@ -0,0 +1,101 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -passes=loop-vectorize -debug-only=loop-vectorize \
+; RUN:   -scalable-vectorization=on -force-target-supports-scalable-vectors \
+; RUN:   -force-tail-folding-style=none  -enable-csa-vectorization \
+; RUN:   -prefer-predicate-over-epilogue=predicate-else-scalar-epilogue \
+; RUN:   -disable-output 2>&1 < %s | FileCheck %s
+
+; This function is generated from the following C/C++ program:
+; int simple_csa_int_select(int N, int *data, int a) {
+;   int t = -1;
+;   for (int i = 0; i < N; i++) {
+;     if (a < data[i])
+;       t = data[i];
+;   }
+;   return t; // use t
+; }
+define i32 @simple_csa_int_select(i32 %N, ptr %data, i64 %a) {
+entry:
+  %cmp9 = icmp sgt i32 %N, 0
+  br i1 %cmp9, label %loop.preheader, label %exit
+
+loop.preheader:                               ; preds = %entry
+  %wide.trip.count = zext i32 %N to i64
+  br label %loop
+
+exit:                                 ; preds = %loop, %entry
+  %t.0.lcssa = phi i32 [ -1, %entry ], [ %spec.select, %loop ]
+  ret i32 %t.0.lcssa
+
+loop:                                         ; preds = %loop.preheader, %loop
+  %iv = phi i64 [ 0, %loop.preheader ], [ %iv.next, %loop ]
+  %t.010 = phi i32 [ -1, %loop.preheader ], [ %spec.select, %loop ]
+  %arrayidx = getelementptr inbounds i32, ptr %data, i64 %iv
+  %0 = load i32, ptr %arrayidx, align 4
+  %1 = sext i32 %0 to i64
+  %cmp1 = icmp slt i64 %a, %1
+  %spec.select = select i1 %cmp1, i32 %0, i32 %t.010
+  %iv.next = add nuw nsw i64 %iv, 1
+  %exitcond.not = icmp eq i64 %iv.next, %wide.trip.count
+  br i1 %exitcond.not, label %exit, label %loop
+}
+
+; CHECK: VPlan 'Initial VPlan for VF={vscale x 1},UF>=1' {
```

Comment:
How did the auto-generated CHECK lines end up after the function? I think the VPlan-based testing can be made more extensive.

Reply:
Sorry, this was not actually auto generated. I updated the file to remove the first line. Would you like to see any other CHECK lines in this file? If so, what would you like to see specifically?

Reply:
> Is it possible to also include an EVL test?

Done

> I think VPlan-based testing can be made more useful by also forcing different VFs?

This seems redundant to me. It should not matter much for the VPlan. 

> I think we can also report instruction costs, since the cost-computation in the code is quite non-trivial.

I'll add another test here for this. I think it could be a good idea to have this depend on forced VFs.